src/topology/uniform_space/equiv.lean

```diff
@@ -0,0 +1,269 @@
+
+namespace uniform_equiv
+variables [uniform_space α] [uniform_space β] [uniform_space γ] [uniform_space δ]
+
+instance : has_coe_to_fun (α ≃ᵤ β) (λ _, α → β) := ⟨λe, e.to_equiv⟩
+
+@[simp] lemma uniform_equiv_mk_coe (a : equiv α β) (b c) :
+  ((uniform_equiv.mk a b c) : α → β) = a :=
+rfl
+
+/-- Inverse of a uniform isomorphism. -/
+protected def symm (h : α ≃ᵤ β) : β ≃ᵤ α :=
+{ uniform_continuous_to_fun  := h.uniform_continuous_inv_fun,
+  uniform_continuous_inv_fun := h.uniform_continuous_to_fun,
+  to_equiv := h.to_equiv.symm }
+
+/-- See Note [custom simps projection]. We need to specify this projection explicitly in this case,
+  because it is a composition of multiple projections. -/
+def simps.apply (h : α ≃ᵤ β) : α → β := h
+/-- See Note [custom simps projection] -/
+def simps.symm_apply (h : α ≃ᵤ β) : β → α := h.symm
+
+initialize_simps_projections uniform_equiv
+  (to_equiv_to_fun → apply, to_equiv_inv_fun → symm_apply, -to_equiv)
+
+@[simp] lemma coe_to_equiv (h : α ≃ᵤ β) : ⇑h.to_equiv = h := rfl
+@[simp] lemma coe_symm_to_equiv (h : α ≃ᵤ β) : ⇑h.to_equiv.symm = h.symm := rfl
+
+lemma to_equiv_injective : function.injective (to_equiv : α ≃ᵤ β → α ≃ β)
+| ⟨e, h₁, h₂⟩ ⟨e', h₁', h₂'⟩ rfl := rfl
+
+@[ext] lemma ext {h h' : α ≃ᵤ β} (H : ∀ x, h x = h' x) : h = h' :=
+to_equiv_injective $ equiv.ext H
+
+/-- Identity map as a uniform isomorphism. -/
+@[simps apply {fully_applied := ff}]
+protected def refl (α : Type*) [uniform_space α] : α ≃ᵤ α :=
+{ uniform_continuous_to_fun := uniform_continuous_id,
+  uniform_continuous_inv_fun := uniform_continuous_id,
+  to_equiv := equiv.refl α }
+
+/-- Composition of two uniform isomorphisms. -/
+protected def trans (h₁ : α ≃ᵤ β) (h₂ : β ≃ᵤ γ) : α ≃ᵤ γ :=
+{ uniform_continuous_to_fun  := h₂.uniform_continuous_to_fun.comp h₁.uniform_continuous_to_fun,
+  uniform_continuous_inv_fun := h₁.uniform_continuous_inv_fun.comp h₂.uniform_continuous_inv_fun,
+  to_equiv := equiv.trans h₁.to_equiv h₂.to_equiv }
+
+@[simp] lemma trans_apply (h₁ : α ≃ᵤ β) (h₂ : β ≃ᵤ γ) (a : α) : h₁.trans h₂ a = h₂ (h₁ a) := rfl
+
+@[simp] lemma uniform_equiv_mk_coe_symm (a : equiv α β) (b c) :
+  ((uniform_equiv.mk a b c).symm : β → α) = a.symm :=
+rfl
+
+@[simp] lemma refl_symm : (uniform_equiv.refl α).symm = uniform_equiv.refl α := rfl
+
+protected lemma uniform_continuous (h : α ≃ᵤ β) : uniform_continuous h :=
+h.uniform_continuous_to_fun
+
+@[continuity]
+protected lemma continuous (h : α ≃ᵤ β) : continuous h :=
+h.uniform_continuous.continuous
+
+protected lemma uniform_continuous_symm (h : α ≃ᵤ β) : uniform_continuous (h.symm) :=
+h.uniform_continuous_inv_fun
+
+@[continuity] -- otherwise `by continuity` can't prove continuity of `h.to_equiv.symm`
+protected lemma continuous_symm (h : α ≃ᵤ β) : continuous (h.symm) :=
+h.uniform_continuous_symm.continuous
+
+/-- A uniform isomorphism as a homeomorphism. -/
+@[simps]
+protected def to_homeomorph {α : Type*} [uniform_space α] (e : α ≃ᵤ α) : α ≃ₜ α :=
```

Comment:
Why do you restrict this definition to `α = β`?

Reply:
Otherwise LGTM but I want to have another look before merging.

Reply:
I don't know either, this looks like a bad copy-paste :sweat_smile: